### ISSUE TITLE
Area Humidity sensor doesn't get created

### ISSUE THREAD

**Issue by mihsu81** [jseidl/hass-magic_areas#208]
Hi @jseidl,

Looks like Magic Areas doesn't create Area Humidity sensors. Is it by design?
I made sure the _device_class_ is set to _humidity_ and _Aggregate Min Entities_ is 1, since I only have 1 sensor per room/area.
I've tried with both Magic Areas 2.0.5 and 3.0.0-beta2.

Thanks in advance.

**Comment by jseidl** [comment]
Hi. It should. Is the humidity sensor right, numeral, not the binary moisture one (just checking)

**Comment by mihsu81** [comment]
Hi @jseidl,
I've realised now that it only affects one area. After deleting the "Magic Areas" integration for that specific area and restarting HA, it got recreated, but now it's also missing _Battery, Energy, Illuminance, Plug, Power [W], Signal Strength [dBm], Temperature [°C], Voltage [V]_, even though they were present yesterday.
The same devices are assigned to the area. 

I guess it's related to the below error.

```
Logger: homeassistant.components.sensor
Source: custom_components/magic_areas/sensor.py:75
Integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
First occurred: 10:09:15 AM (3 occurrences)
Last logged: 11:13:10 AM

Error while setting up magic_areas platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/magic_areas/sensor.py", line 32, in async_setup_entry
    await load_sensors(hass, async_add_entities, area)
  File "/config/custom_components/magic_areas/sensor.py", line 75, in load_sensors
    device_class, unit_of_measurement = map_key.split("/")
ValueError: too many values to unpack (expected 2)
```

**Comment by mihsu81** [comment]
Hi @jseidl,

Looks like the issue is caused by a Xiaomi Mi Air Purifier 3C (zhimi.airpurifier.mb4) added through the _Xiaomi Miio_ integration.
When I remove it from the zone all Area sensors get created and updated properly.

You can find it in the debug log with friendly name _SONIAS-BEDROOM-AIR-PURIFIER_ and its entities contain _sonias_bedroom_air_purifier_.

**Comment by jseidl** [comment]
That's good to know. Can you post a screenshot of this entity on the developer tools listing? That would show me all the attributes and stuff that might be screwing up.

What I've been noticing is that if the entity has an attribute called `entity_id`, it screws up (like groups for eg)

**Comment by mihsu81** [comment]
There are multiple entities assigned to the device.
![image](https://user-images.githubusercontent.com/40715497/165683990-e22cb914-bfe4-4f1f-bc54-56c1a177a98f.png)
![image](https://user-images.githubusercontent.com/40715497/165684054-204b4ed5-1277-4098-9020-43664bbb7ff8.png)
![image](https://user-images.githubusercontent.com/40715497/165684094-4b2fd80e-b438-4414-b3dd-01cd891d99ea.png)
![image](https://user-images.githubusercontent.com/40715497/165684156-b52d4194-a812-4a0f-afa4-a4ca7543dead.png)
![image](https://user-images.githubusercontent.com/40715497/165684217-e77e6d9d-6fd2-428f-9e64-9e83783abf4d.png)
![image](https://user-images.githubusercontent.com/40715497/165684644-f2b674b7-5dc0-423f-b1c1-c6a2ae09b76f.png)
![image](https://user-images.githubusercontent.com/40715497/165684842-0748ce6c-f319-48ad-972b-82cc7ad2e653.png)
![image](https://user-images.githubusercontent.com/40715497/165684878-065c1579-239d-4bf5-80bf-ac18001aef28.png)
![image](https://user-images.githubusercontent.com/40715497/165684914-b8c792b7-25bd-4cc8-9fa4-75747ca2b2c4.png)

**Comment by jseidl** [comment]
Hmmm I don't see anything special with those entities. Might be related to another bug I'm working on, will let you know once I have something!

**Comment by jseidl** [comment]
LOL I just saw it. 

I was doing something very poorly on resolving the device_class / unit_of_measurement pair. Anyway. Rewrote the whole function, should be good now.

Haven't tested yet, I'm mid working on another bug as well but feel free to try out code from the branche of #211

**Comment by mihsu81** [comment]
Hi @jseidl,
Sorry for not getting back to you sooner. 
I've added your change but it seems to throw additional errors and some aggregated sensors are not created anymore: Battery, Energy, Frequency, Humidity, Illuminance, Power, Signal Strength, Temperature, Voltage.


```
Log Details (ERROR)
Logger: homeassistant.components.sensor
Source: custom_components/magic_areas/sensor.py:63
Integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
First occurred: 6:01:24 PM (7 occurrences)
Last logged: 6:01:25 PM

Error while setting up magic_areas platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/magic_areas/sensor.py", line 32, in async_setup_entry
    await load_sensors(hass, async_add_entities, area)
  File "/config/custom_components/magic_areas/sensor.py", line 63, in load_sensors
    device_class_uom_pairs.append(entity['device_class'], entity['unit_of_measurement'])
TypeError: list.append() takes exactly one argument (2 given)
```

**Comment by jseidl** [comment]
Yeah this branch accidentally got broken, sorry about that, I'll push a new one with the fix

**Comment by jseidl** [comment]
Fixed on 3.0. If the issue persists, please reopen!
